espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/drivers/ComposeDriver.kt

```diff
@@ -29,22 +29,31 @@ class ComposeDriver : AppDriver {
     override val name = DriverContext.StrategyType.COMPOSE
 
     override fun findElement(params: Locator): BaseElement {
-        val parentNodeInteraction = params.elementId?.let { getNodeInteractionById(it) }
-
-        val nodeInteractions = parentNodeInteraction?.findDescendantNodeInteractions(params)
-            ?: EspressoServerRunnerTest.composeTestRule.onAllNodes(semanticsMatcherForLocator(params))
-
-        if (nodeInteractions.fetchSemanticsNodes(false).isEmpty()) {
-            throw NoSuchElementException(
-                String.format(
-                    "Could not find a compose element with strategy '%s' and selector '%s'",
-                    params.using, params.value
-                )
+        val nodeInteractions = nodeInteractionCollection(params)
+        if (nodeInteractions.fetchSemanticsNodes(false).isEmpty()) throw NoSuchElementException(
+            String.format(
+                "Could not find a compose element with strategy '%s' and selector '%s'",
+                params.using, params.value
             )
-        }
+        )
         return ComposeElement(nodeInteractions[0])
     }
 
+    override fun findElements(params: Locator): List<BaseElement> {
+        val nodeInteractions = nodeInteractionCollection(params)
+        val elements = mutableListOf<BaseElement>()
```

Comment:
`return nodeInteractions.fetchSemanticsNodes(false). map { ComposeElement(nodeInteractions[it]) }`

Reply:
`it` inside is not integer index, but `SemanticsNode`. 
Something like this could work:
```
override fun findElements(params: Locator): List<BaseElement> {
        val nodeInteractions = toNodeInteractionsCollection(params)
        return nodeInteractions.fetchSemanticsNodes(false)
            .mapIndexed { index, semanticsNode -> ComposeElement(nodeInteractions[index]) }
    }
```

Reply:
Works for me

Reply:
changed